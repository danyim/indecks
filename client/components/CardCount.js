import React from 'react';
import styles from '../styles/components/CardCount';

const CardCount = React.createClass({
  render() {
    const { current, max } = this.props;
    console.log(styles);
    return (
      <div>
        <span className={styles.current}>{current}</span>
        <span className={styles.separator}>of</span>
        <span className={styles.max}>{max}</span>
      </div>
    );
  }
});

export default CardCount;
