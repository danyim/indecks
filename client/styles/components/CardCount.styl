@import '../_variables';

.cardCount
  display inline-block
  text-align right
  color grey
  // All the other font sizes are in em, which means this value
  // will determine the scale
  font-size 0.8rem

.current
  color blue
  font 400 3em 'Dosis'
  position relative
  top -0.5em
  right 0em

.separator
  font 200 4em 'Dosis'

.max
  font 200 1.5em 'Dosis'
  position relative
  top -0.5em

