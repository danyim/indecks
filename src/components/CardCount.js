import React from 'react';
import styles from '../styles/components/CardCount';

class CardCount extends React.Component {
  render() {
    const { current, max } = this.props;
    return (
      <div className={styles.cardCount}>
        <span className={styles.current}>{current}</span>
        <span className={styles.separator}>/</span>
        <span className={styles.max}>{max}</span>
      </div>
    );
  }
}

export default CardCount;
