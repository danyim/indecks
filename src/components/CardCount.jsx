import React from 'react';
import styles from '../styles/components/CardCount';

const propTypes = {
  current: React.PropTypes.number.isRequired,
  max: React.PropTypes.number.isRequired
};

const defaultProps = {};

const CardCount = (props) => {
  const { current, max } = props;
  return (
    <div className={styles.cardCount}>
      <span className={styles.current}>{current}</span>
      <span className={styles.separator} />
      <span className={styles.max}>{max}</span>
    </div>
  );
};

CardCount.propTypes = propTypes;
CardCount.defaultProps = defaultProps;

export default CardCount;
