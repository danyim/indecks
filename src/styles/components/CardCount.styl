@import '../_variables';

.cardCount
  display inline-block
  text-align right
  color grey
  // All the other font sizes are in em, which means this value
  // will determine the scale
  font-size 0.8rem

.current
  color blue
  font 400 3em 'Dosis'
  position relative
  top -0.5em
  right 0em

.separator:after
  font 200 4em 'Dosis'
  content "/"

.max
  font 200 1.5em 'Dosis'
  position relative
  top -0.5em

@media (max-width: cardWidth)
  .current, .separator:after, .max
    font-size: 3em
    top 0
    right 0
  .separator:after
    content "|"
    margin 0 1.0rem

