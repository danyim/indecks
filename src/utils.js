export const createReducer = (initialState, handlers) => {
  return function reducer(state = initialState, action = {}) {
    if (handlers.hasOwnProperty(action.type)) {
      return handlers[action.type](state, action);
    }
    return state;
  };
};

/**
 * Given an array of consecutive numbers, find the missing value.
 * Returns null if no missing value found
 * @param  {Array}  a Array of numbers
 * @return {Number}   Missing value
 */
export const findMissing = (a) => {
  let expectedNextValue;
  for(let k of a)
  {
    expectedNextValue = a[k] + 1;
    if(k <= a.length - 1 && expectedNextValue !== a[k + 1]) {
      return expectedNextValue;
    }
  }
  return null;
};

/**
 * Given an array with potentially missing values at parameter key, find the
 * next best 0-based index to insert the next value
 * @param  {Array}  a   An array of objects
 * @param  {String} key Key to look for
 * @return {Number}     Index of the best next value
 */
export const findBestNextIndex = (a, key) => {
  return findMissing(a.map(x => x[key])) || Math.max(0, Math.max.apply(null, a.map(x => x[key])) + 1);
};

