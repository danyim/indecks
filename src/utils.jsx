import React from 'react'
import { Route } from 'react-router-dom'
import _ from 'lodash'

export const createReducer = (initialState, handlers) =>
  function reducer(state = initialState, action = {}) {
    if (Object.prototype.hasOwnProperty.call(handlers, action.type)) {
      return handlers[action.type](state, action)
    }
    return state
  }

/**
 * Given an array of consecutive numbers from 1 to n, find the missing value.
 * Returns null if no missing value found
 * @param  {Array}  a   Array of numbers
 * @param  {Number} min Minimum value (0 or 1)
 * @return {Number}     Missing value
 */
export const findMissing = (a, min = 1) => {
  let expected = min
  for (let k = 0; k < a.length; k++) {
    if (a[k] !== expected) {
      return expected
    }
    expected++
  }
  return null
}

/**
 * Given an array of objects, check the value of the key and return the
 * best 1-based index to insert the next value
 * @param  {Array}  a   An array of objects
 * @param  {String} key Key to look for
 * @return {Number}     Index of the best next value
 */
export const findBestNextIndex = (a, key) =>
  findMissing(a.map(x => x[key])) ||
  Math.max(0, Math.max.apply(null, a.map(x => x[key])) + 1)

/**
 * Helper function to generate routes
 * @param  {Object} route) Route object that takes the form
 *                         {
 *                           exact: boolean,
 *                           path: string,
 *                           component: Component,
 *                           routes: Routes
 *                         }
 * @return {Component}        Returns a <Route> component to plug into a <Switch>
 */
export const routeWithSubRoutes = route =>
  <Route
    key={_.uniqueId()}
    exact={route.exact || false}
    path={route.path}
    render={props =>
      // Pass the sub-routes down to keep nesting
      // <route.component {...props}>
      //   { route.children ? route.children.map(child => routeWithSubRoutes(child)) : null }
      // </route.component>
      <route.component {...props} routes={route.routes} />}
  />
