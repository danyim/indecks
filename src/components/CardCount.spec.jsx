import toJson from 'enzyme-to-json'
import { shallow, mount, setup, setupFull } from '../test/utils' // eslint-disable-line no-unused-vars
import CardCount from './CardCount'

const defaultProps = {
  current: 25,
  max: 100,
}

describe('CardCount', () => {
  it('should render self and subcomponents', () => {
    const wrapper = mount(CardCount, defaultProps)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should render the correct card current/max', () => {
    const { wrapper } = setup(CardCount, defaultProps)
    expect(wrapper.find('span.current').text()).toEqual(
      `${defaultProps.current}`
    )
    expect(wrapper.find('span.max').text()).toEqual(`${defaultProps.max}`)
  })
})
