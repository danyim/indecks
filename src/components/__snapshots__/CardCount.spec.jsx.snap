// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardCount should render self and subcomponents 1`] = `
<CardCount
  current={25}
  max={100}
>
  <div>
    <span>
      25
    </span>
    <span />
    <span>
      100
    </span>
  </div>
</CardCount>
`;
